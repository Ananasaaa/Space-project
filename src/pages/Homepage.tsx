import { mockImage } from '../mocks/mockImg';
import '../styles/homepage.scss';

const backgroundStyle = {
  backgroundImage: `url(${process.env.PUBLIC_URL}/img/bg_homepage.jpg)`,
  backgroundSize: 'cover',
  backgroundPosition: 'center',
  backgroundRepeat: 'no-repeat',
};

const Homepage = () => {
  return (
    <div className="homepage" style={backgroundStyle}>
      <h1>Space Image of the Day Viewer</h1>
      <div className="image-container">
        <h2>{mockImage.title}</h2>
        <p>{mockImage.date}</p>
        <img src={mockImage.url} alt={mockImage.title} className="image" />
        <p className="description">{mockImage.explanation}</p>
      </div>
    </div>
  );
};

export default Homepage;
