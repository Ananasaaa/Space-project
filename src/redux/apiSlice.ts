import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

export const nasaApi = createApi({
  reducerPath: 'nasaApi',
  baseQuery: fetchBaseQuery({
    baseUrl: 'https://api.nasa.gov/',
  }),
  endpoints: (builder) => ({
    getSpaceImage: builder.query({
      query: () => ({
        url: 'planetary/apod',
        params: {
          api_key: process.env.REACT_APP_NASA_KEY as string,
        },
      }),
    }),
    getMarsPhotos: builder.query({
      query: ({ sol = 4000, camera, page = 1 }) => ({
        url: 'mars-photos/api/v1/rovers/curiosity/photos',
        params: {
          sol,
          page,
          ...(camera ? { camera } : {}),
          api_key: process.env.REACT_APP_NASA_KEY as string,
        },
      }),
    }),
  }),
});

export const { useGetSpaceImageQuery, useGetMarsPhotosQuery } = nasaApi;
