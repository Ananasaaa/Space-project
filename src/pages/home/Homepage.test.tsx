import { render } from '@testing-library/react';
import Homepage from './index';
import { useGetSpaceImageQuery } from '../../redux/apiSlice';

jest.mock('../../redux/apiSlice');

describe('Homepage component', () => {
  it('should render the title, date, image, and description without redux store', () => {
    (useGetSpaceImageQuery as jest.Mock).mockReturnValue({
      data: {
        title: 'Test Space Image',
        date: '2024-04-26',
        url: 'https://example.com/image.jpg',
        explanation: 'Test explanation about the space image.',
      },
      isLoading: false,
      error: null,
    });

    const { asFragment } = render(<Homepage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
