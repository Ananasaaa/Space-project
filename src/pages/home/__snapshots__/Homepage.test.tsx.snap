// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component should render the title, date, image, and description without redux store 1`] = `
<DocumentFragment>
  <div
    class="homepage"
    style="background-image: url(/img/bg_homepage.jpg); background-size: cover; background-position: center; background-repeat: no-repeat;"
  >
    <h1>
      Space Image of the Day Viewer
    </h1>
    <div
      class="image-container fade-in"
    >
      <h2>
        Test Space Image
      </h2>
      <p>
        2024-04-26
      </p>
      <img
        alt="Test Space Image"
        class="image"
        loading="lazy"
        src="https://example.com/image.jpg"
      />
      <p
        class="description"
      >
        Test explanation about the space image.
      </p>
    </div>
    <button
      class="gallery-button"
    >
      See last 7 days
    </button>
  </div>
</DocumentFragment>
`;
